analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/IOperationExtensionsTest.cs

```diff
@@ -32,18 +32,32 @@ public class IOperationExtensionsTest
         [TestMethod]
         public void TrackedSymbol_LocalReference_IsVariableSymbol()
         {
-            var localreference = ((ISimpleAssignmentOperation)TestHelper.CompileCfgBodyCS("var a = true;").Blocks[1].Operations[0]).Target;
-            var symbol = ILocalReferenceOperationWrapper.FromOperation(localreference).Local;
-            localreference.TrackedSymbol().Should().Be(symbol);
+            var localReference = ((ISimpleAssignmentOperation)TestHelper.CompileCfgBodyCS("var a = true;").Blocks[1].Operations[0]).Target;
+            var symbol = ILocalReferenceOperationWrapper.FromOperation(localReference).Local;
+            localReference.TrackedSymbol().Should().Be(symbol);
         }
 
         [TestMethod]
         public void TrackedSymbol_ParameterReference_IsParameterSymbol()
         {
             var expressionStatement = (IExpressionStatementOperation)TestHelper.CompileCfgBodyCS("parameter = true;", "bool parameter").Blocks[1].Operations[0];
-            var paramaterReference = ((ISimpleAssignmentOperation)expressionStatement.Operation).Target;
-            var symbol = IParameterReferenceOperationWrapper.FromOperation(paramaterReference).Parameter;
-            paramaterReference.TrackedSymbol().Should().Be(symbol);
+            var parameterReference = ((ISimpleAssignmentOperation)expressionStatement.Operation).Target;
+            var symbol = IParameterReferenceOperationWrapper.FromOperation(parameterReference).Parameter;
+            parameterReference.TrackedSymbol().Should().Be(symbol);
+        }
+
+        [DataTestMethod]
+        [DataRow(@"public class C { int a = 2; void Method() { a = 1; } }")]
+        [DataRow(@"public class C { int a = 2; void Method() { this.a = 1; } }")]
+        [DataRow(@"public class C { static int StaticField = 2; void Method() { StaticField = 1; } }")]
+        [DataRow(@"public class C { static int StaticField = 2; void Method() { C.StaticField = 1; } }")]
```

Comment:
_Optional:_ We don't need to initialize the fields. And we can scaffold the class inside the method. Like that, we can have only the `assignment` as a parameter.
```
var code = $"public class C {{ int field; static int StaticField`; void Method() {{ {assignment}; }} }}";
```

Reply:
I've also renamed `a` to `field` for consistency because we have the `StaticField` as a counterpart.